app.py

```diff
@@ -1,14 +1,92 @@
 from flask import Flask, request, render_template
 import redis
+import psycopg2
+from psycopg2.extensions import AsIs
 import json
 import logging
 
 logger = logging.getLogger(__name__)
 
-r = redis.Redis(host='172.17.0.1', port=6379)
-
 app = Flask(__name__)
 
+def initialize():
+    global p
+    global r
+    try:
+        p = psycopg2.connect(user="postgres",
+                            password="postgres",
+                            host="172.17.0.1",
+                            port="5432",
+                            database="hello")
+
+        r = redis.Redis(host='172.17.0.1', port=6379)
+    except (Exception) as e:
+        return False
+    return True
+
+@app.route("/users", methods = ['POST', 'GET'])
+def users():
+
+    # If both are specified, id takes precedence
+    if request.method == 'GET':
+
+        c = p.cursor()
+
+        id   = request.args.get('id')
+        name = request.args.get('name')
+
+        try:
+            if id:
+                c.execute("SELECT * FROM users WHERE id = %s;", (id,))
+                record = c.fetchmany()
+            elif name and not id:
+                c.execute("SELECT * FROM users WHERE name = %s;", (name,))
+                record = c.fetchmany()
+            elif not id and not name:
+                c.execute("SELECT * FROM users")
+                record = c.fetchall()
+        except (Exception) as e:
+            return 'Could not connect to database', 503
+            exit(1)
+
+        if not record:
+            return_string = ""
+        else:
+            return_string = ""
+            for entry in record:
+                entry_string='{"id":"'      + str(entry[0]) + \
+                           '", "name":"'    + entry[1] + \
+                           '", "address":"' + entry[2] + \
+                           '", "phone":"'   + entry[3] + '"}'
+                return_string += entry_string + '\n'
+
+        return '\n' + return_string + '\n'
+        c.close()
+
+    if request.method == 'POST':
+        e = ""
+        # curl -X POST -H "Content-Type:application/json" -d '{"name":"Chris", "address":"2345 DNA Dumpster Lane", "phone":"123-456-7890"}' localhost:5000/users
+        d = request.get_json()
+        if not d:
+            return 'JSON POST failure', 400
+        else:
+            columns = d.keys()
+            values = [d[column] for column in columns]

```

Comment:
Careful, you're not sanitizing inputs here.  Perhaps its fine in this case, but what if there was a column I *wasn't* supposed to be able to modify via a POST request, but I had it in the json.  It would get set based on this.

This is actually a really clever way to do it, and I like it.  But its currently not safe.  You need to make sure you *only* set `name`, `address` and `phone`

Reply:
As a side note, this is the sort of shit hackers look for when they hit endpoints.  They intentionally do shit to see if there are unsanitized DB queries.

Reply:
Yea, it may be clever, but its not mine. you're assuming I have intelligence wrapped up in how this works.  I "got it running" and figured it was probably a cluster, so let's discuss a better way to do it.